airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -492,6 +492,18 @@ public void resetConnection() {
     }
   }
 
+  @Override
+  public void resetConnectionAndSkipNextScheduling() {
+    if (workflowState.isDoneWaiting()) {
+      workflowState.setCancelledForReset(true);
```

Comment:
For my understanding, `setCancelledForReset` means that we cancel the current run and start a reset?